sky/execution.py

```diff
@@ -181,9 +188,13 @@ def _execute(
             f'{colorama.Style.RESET_ALL}')
 
     cluster_exists = False
+    existing_config_hash = None
     if cluster_name is not None:
         cluster_record = global_user_state.get_cluster_from_name(cluster_name)
-        cluster_exists = cluster_record is not None
+        if cluster_record is not None:
+            cluster_exists = True
+            if skip_unecessary_provisioning:
+                existing_config_hash = cluster_record['config_hash']
```

Comment:
For thread safety should we move the config hash retrieving and comparing into the `provision` function, so that they can be wrapped in the lock?

Reply:
Also, in that way, we can make the argument a bit higher level, e.g., `skip_if_no_updates: bool`